pyhf/constraints.py

```diff
@@ -1,5 +1,36 @@
 from . import get_backend, default_backend
 
+
+class standard_gaussian_constraint(object):
+    def __init__(self, n_parameters, inits, bounds, auxdata):
+        self.pdf_type = 'normal'
+        self.n_parameters = n_parameters
+        self.suggested_init = inits
+        self.suggested_bounds = bounds
+        self.auxdata = auxdata
+
+    def alphas(self, pars):
+        return pars  # the nuisance parameters correspond directly to the alpha
+
+    def expected_data(self, pars):
+        return self.alphas(pars)
+
+class factor_poisson_constraint(object):
+    def __init__(self, n_parameters, inits, bounds, auxdata, factors):
+        self.factors = factors
+        self.pdf_type = 'poisson'
+        self.n_parameters = n_parameters
+        self.suggested_init = inits
+        self.suggested_bounds = bounds
+        self.auxdata = auxdata
+
+    def alphas(self, pars):
```

Comment:
need a comment on why you're doing the product/stack thing (maybe explain the math here)